frontend/src/v5/ui/routes/viewer/tickets/ticket.context.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ *  Copyright (C) 2023 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { createContext, useEffect, useState } from 'react';
+
+export enum TicketDetailsView {
+	Form,
+	Groups,
+}
+
+export interface TicketContextType {
+	view: TicketDetailsView;
+	viewProps?: any;
+	setDetailViewAndProps: (view: TicketDetailsView, props?: any) => void;
+}
+
+const defaultValue: TicketContextType = { view: TicketDetailsView.Form, setDetailViewAndProps: () => {} };
+export const TicketContext = createContext(defaultValue);
+TicketContext.displayName = 'TicketContext';
+
+export const TicketContextComponent = ({ children }) => {
+	const [view, setView] = useState(TicketDetailsView.Form);
+	const [viewProps, setViewProps] = useState();
+	const [contextValue, setContextValue] = useState<TicketContextType>({ view, setDetailViewAndProps: () => null });
+
+	useEffect(() => {
+		setContextValue({ view,
+			viewProps,
+			setDetailViewAndProps: (viewParam, props) => {
+				if (props) {
+					setViewProps(props);
+				}
+				setView(viewParam);
+			},
+		});
+	}, [view, viewProps]);
+
+	return (
+		<TicketContext.Provider value={contextValue}>
+			{children}
+		</TicketContext.Provider>
+	);
+};
```

Comment:
I think the `useEffect` and the `context/setContext` are not necessary, instead, you can pass `view, viewProps, setDetailViewAndProps` directly to the contextProvider (which imo, makes the code a little cleaner) like this 
```ts
export const TicketContextComponent = ({ children }) => {
	const [view, setView] = useState(TicketDetailsView.Form);
	const [viewProps, setViewProps] = useState();

	const setDetailViewAndProps = (viewParam, props) => {
		if (props) {
			setViewProps(props);
		}
		setView(viewParam);
	};

	return (
		<TicketContext.Provider value={{ view, viewProps, setDetailViewAndProps }}>
			{children}
		</TicketContext.Provider>
	);
};
```